src/common/dao/project.go

```diff
@@ -180,99 +181,102 @@ func SearchProjects(userID int) ([]models.Project, error) {
-func GetUserRelevantProjects(userID int, projectName string, args ...int64) ([]models.Project, error) {
-	return getProjects(userID, projectName, args...)
+// GetProjects returns a project list according to the query conditions
+func GetProjects(owner, name, public, member string,
+	role int, page, size int64) ([]*models.Project, error) {
+	sql, params := queryConditions(owner, name, public, member, role)
+
+	sql = `select distinct p.project_id, p.name, p.public, p.owner_id, 
+				p.creation_time, p.update_time ` + sql
+	if size > 0 {
+		sql += ` limit ?`
+		params = append(params, size)
+
+		if page > 0 {
+			sql += ` offset ?`
+			params = append(params, (page-1)*size)
+		}
+	}
+
+	var projects []*models.Project
+	_, err := GetOrmer().Raw(sql, params).QueryRows(&projects)
+	return projects, err
 }
 
-// GetTotalOfProjects returns the total count of projects
-func GetTotalOfProjects(name string, public ...int) (int64, error) {
-	qs := GetOrmer().
-		QueryTable(new(models.Project)).
-		Filter("Deleted", 0)
+func queryConditions(owner, name, public, member string,
+	role int) (string, []interface{}) {
+	params := []interface{}{}
 
-	if len(name) > 0 {
-		qs = qs.Filter("Name__icontains", name)
-	}
+	sql := ` from project p`
 
-	if len(public) > 0 {
-		qs = qs.Filter("Public", public[0])
+	if len(owner) != 0 {
+		sql += ` join user u1
+					on p.owner_id = u1.user_id`
 	}
 
-	return qs.Count()
-}
-
-// GetProjects returns project list
-// args[0]: public, args[1]: limit, args[2]: offset
-func GetProjects(name string, args ...int64) ([]models.Project, error) {
-	return getProjects(0, name, args...)
-}
-
-func getProjects(userID int, name string, args ...int64) ([]models.Project, error) {
-	projects := []models.Project{}
+	if len(member) != 0 {
+		sql += ` join project_member pm
+					on p.project_id = pm.project_id
+					join user u2
+					on pm.user_id=u2.user_id`
+	}
+	sql += ` where p.deleted=0`
 
-	o := GetOrmer()
-	sql := ""
-	queryParam := []interface{}{}
-
-	if userID != 0 { //get user's projects
-		sql = `select distinct p.project_id, p.owner_id, p.name, 
-					p.creation_time, p.update_time, p.public, pm.role role 
-			from project p 
-			left join project_member pm 
-			on p.project_id = pm.project_id
-	 		where p.deleted = 0 and pm.user_id= ?`
-		queryParam = append(queryParam, userID)
-	} else { // get all projects
-		sql = `select * from project p where p.deleted = 0 `
+	if len(owner) != 0 {
+		sql += ` and u1.username=?`
+		params = append(params, owner)
 	}
 
-	if name != "" {
-		sql += ` and p.name like ? `
-		queryParam = append(queryParam, "%"+escape(name)+"%")
+	if len(name) != 0 {
+		sql += ` and p.name like ?`
+		params = append(params, "%"+escape(name)+"%")
 	}
 
-	switch len(args) {
-	case 1:
-		sql += ` and p.public = ?`
-		queryParam = append(queryParam, args[0])
-		sql += ` order by p.name `
-	case 2:
-		sql += ` order by p.name `
-		sql = paginateForRawSQL(sql, args[0], args[1])
-	case 3:
+	if len(public) != 0 {
```

Comment:
what about if I set public as "somevalue"? As it may come from Rest, the check if not enough I think.